services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/oauth/OAuth2Api.java

```diff
@@ -88,6 +96,35 @@ public OAuth2Service createService(OAuthConfig config) {
         return new BasicOAuth2Service(this, config);
     }
         
+    static ProvidedUserInfo parseUserInfo(String body) {
+    	try {
+			JSONObject json = new JSONObject(body);
+
+			ProvidedUserInfo info = new ProvidedUserInfo();
+			if (json.has(FAMILY_NAME)) {
+				info.setLastName(json.getString(FAMILY_NAME));
+			}
+			if (json.has(GIVEN_NAME)) {
+				info.setFirstName(json.getString(GIVEN_NAME));
+			}
+			if (json.has(SUB)) {
+				info.setSubject(json.getString(SUB));
+			}
+			if (json.has(EMAIL)) {
+				info.setEmail(json.getString(EMAIL));
+			}
+			if (json.has(EMAIL_VERIFIED)) {
+				info.setEmailVerified(json.getString(EMAIL_VERIFIED));

```

Comment:
Should ProvidedUserInfo.emailVerified be a boolean, and use `json.getBoolean(EMAIL_VERIFIED)` here?

Reply:
No, because i'm not sure what Okta puts into the `email_verified` string and I want to capture the raw data even if it isn't literally `true` or `false`.  If it's an unexpected value we will return it in the error message added in the previous PR which in turn will help our debugging.

Reply:
Ok.  So if the values is not a true or false, then the `json.getBoolean(EMAIL_VERIFIED)` on line 119 will cause a JSONException (which will be wrapped in an UnauthorizedException)